### ISSUE TITLE
error from pip install awscli "'install_requires' must be a string" on centos7

### ISSUE THREAD

**Issue by davidcallen** [christiangda/ansible-role-awscli#1]
Firstly thanks for this excellent ansible role.

I've been using it for 18 months with no issue but today I now get an error from my Packer/Ansible run  for building AMI on AWS.

Error is 
``error in awscli setup command: 'install_requires' must be a string or list of strings containing valid project``

It ran OK yesterday. I have not changed my version of ansible or packer. These are 

``ansible --version
ansible 2.9.20
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/david.allen/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  executable location = /bin/ansible
  python version = 3.9.4 (default, Apr  6 2021, 00:00:00) [GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]``

``packer --version
1.6.6``

Full log is : 

``    amazon-ebs: TASK [christiangda.awscli : Include variables for RedHat/CentOS] ***************
    amazon-ebs: ok: [default] => (item=/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/redhat-7.yml) => {"ansible_facts": {"cli_dependencies_packages": ["python-pip", "python-pysocks"]}, "ansible_included_var_files": ["/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/redhat-7.yml"], "ansible_loop_var": "item", "changed": false, "item": "/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/redhat-7.yml"}
    amazon-ebs:
    amazon-ebs: TASK [christiangda.awscli : Include variables for Debian Jessie] ***************
    amazon-ebs: skipping: [default] => (item=/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/main.yml)  => {"ansible_loop_var": "item", "changed": false, "item": "/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/main.yml", "skip_reason": "Conditional result was False"}
    amazon-ebs:
    amazon-ebs: TASK [christiangda.awscli : Include variables for Debian (not Jessie)] *********
    amazon-ebs: skipping: [default] => (item=/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/main.yml)  => {"ansible_loop_var": "item", "changed": false, "item": "/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/main.yml", "skip_reason": "Conditional result was False"}
    amazon-ebs:
    amazon-ebs: TASK [christiangda.awscli : Include variables for Ubuntu] **********************
    amazon-ebs: skipping: [default] => (item=/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/main.yml)  => {"ansible_loop_var": "item", "changed": false, "item": "/home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/vars/main.yml", "skip_reason": "Conditional result was False"}
    amazon-ebs:
    amazon-ebs: TASK [christiangda.awscli : Include tasks for installation on RedHat/CentOS] ***
    amazon-ebs: included: /home/david.allen/dev-cloud/host-main/cloud/packer/centos7-glass/ansible/roles/christiangda.awscli/tasks/install-redhat-7.yml for default
    amazon-ebs:
    amazon-ebs: TASK [christiangda.awscli : Check if epel_repo exist for RedHat/CentOS] ********
    amazon-ebs: ok: [default] => {"changed": false, "cmd": "yum repolist 2>/dev/null | grep 'epel' && echo 'true'|| echo 'false'", "delta": "0:00:00.493214", "end": "2021-05-07 12:23:44.746649", "rc": 0, "start": "2021-05-07 12:23:44.253435", "stderr": "", "stderr_lines": [], "stdout": " * epel: d2lzkl7pfhq30w.cloudfront.net\nepel/x86_64           Extra Packages for Enterprise Linux 7 - x86_64      13,589\ntrue", "stdout_lines": [" * epel: d2lzkl7pfhq30w.cloudfront.net", "epel/x86_64           Extra Packages for Enterprise Linux 7 - x86_64      13,589", "true"]}
    amazon-ebs:
    amazon-ebs: TASK [christiangda.awscli : Fail when EPEL repository is not installed] ********
    amazon-ebs: skipping: [default] => {"changed": false, "skip_reason": "Conditional result was False"}
    amazon-ebs:
    amazon-ebs: TASK [christiangda.awscli : Install "['python-pip', 'python-pysocks']" dependencies from EPEL repository] ***
    amazon-ebs: changed: [default] => {"changed": true, "changes": {"installed": ["python-pip", "python-pysocks"]}, "msg": "", "rc": 0, "results": ["Loaded plugins: fastestmirror\nLoading mirror speeds from cached hostfile\n * base: mirror.centos.org\n * epel: d2lzkl7pfhq30w.cloudfront.net\n * extras: mirror.centos.org\n * updates: mirror.centos.org\nResolving Dependencies\n--> Running transaction check\n---> Package python2-pip.noarch 0:8.1.2-14.el7 will be installed\n---> Package python2-pysocks.noarch 0:1.6.8-7.el7 will be installed\n--> Finished Dependency Resolution\n\nDependencies Resolved\n\n================================================================================\n Package                 Arch           Version              Repository    Size\n================================================================================\nInstalling:\n python2-pip             noarch         8.1.2-14.el7         epel         1.7 M\n python2-pysocks         noarch         1.6.8-7.el7          epel          30 k\n\nTransaction Summary\n================================================================================\nInstall  2 Packages\n\nTotal download size: 1.7 M\nInstalled size: 7.3 M\nDownloading packages:\n--------------------------------------------------------------------------------\nTotal                                              3.6 MB/s | 1.7 MB  00:00     \nRunning transaction check\nRunning transaction test\nTransaction test succeeded\nRunning transaction\n  Installing : python2-pysocks-1.6.8-7.el7.noarch                           1/2 \n  Installing : python2-pip-8.1.2-14.el7.noarch                              2/2 \n  Verifying  : python2-pip-8.1.2-14.el7.noarch                              1/2 \n  Verifying  : python2-pysocks-1.6.8-7.el7.noarch                           2/2 \n\nInstalled:\n  python2-pip.noarch 0:8.1.2-14.el7     python2-pysocks.noarch 0:1.6.8-7.el7    \n\nComplete!\n"]}
    amazon-ebs:
    amazon-ebs: TASK [christiangda.awscli : Install awscli] ************************************
    amazon-ebs: fatal: [default]: FAILED! => {"changed": false, "cmd": ["/bin/pip2", "install", "awscli"], "msg": "stdout: Collecting awscli\n  Downloading https://files.pythonhosted.org/packages/ae/8f/f78677c4b8db282145ee29b549a6039aebe71bb6c04a30e3271def395248/awscli-1.19.68.tar.gz (1.4MB)\n    Complete output from command python setup.py egg_info:\n    /usr/lib64/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'\n      warnings.warn(msg)\n    error in awscli setup command: 'install_requires' must be a string or list of strings containing valid project/version requirement specifiers\n    \n    ----------------------------------------\n\n:stderr: Command \"python setup.py egg_info\" failed with error code 1 in /tmp/pip-build-Dsb8Ga/awscli/\nYou are using pip version 8.1.2, however version 21.1.1 is available.\nYou should consider upgrading via the 'pip install --upgrade pip' command.\n"}
    amazon-ebs:``

I notice that it is using pip2 as the executable for installation.
By experimentation I find that when using pip3 it works.

I am not too concerned (I dont think) if it uses pip3.  This may cause issue for others though ...

**Comment by christiangda** [comment]
@davidcallen thanks for your comment, I'll dedicate time to find a solution to this.

**Comment by davidcallen** [comment]
Thanks. I have meant to add that the problem may be intermittent. I need to revisit it and check.
I do have a fork with a quick workaround in it :
https://github.com/davidcallen/ansible-role-awscli

**Comment by christiangda** [comment]
@davidcallen I think I fixed that adding `  - python-setuptools` package to cento-7 dependencies

```yaml
cli_dependencies_packages:
  - python-pip
  - python-pysocks
  - python-setuptools
```

I'm working on develop branch

**Comment by davidcallen** [comment]
Thanks very much for the fix and new version 1.0.1
I have used version 1.0.1 on 6 different Packer EBS (AWS) AMI builds.
No ansible runtime errors found.

I did find an error on my Ubuntu 16.04 AMI build and have raised [Issue #5](https://github.com/christiangda/ansible-role-awscli/issues/5)  including a possible workaround/fix.
